geopandas/vectorized.pyx

```diff
@@ -1075,6 +1099,23 @@ class GeometryArray(object):
         return buffer(self.data, distance, resolution, cap_style, join_style,
                       mitre_limit)
 
+    def types(self):
+        """
+        Types of the underlying Geometries
+
+        Returns
+        -------
+        Pandas categorical with types for each geometry
+        """
+        x = vec_type(self.data)
+
+        types = GEOMETRY_TYPES[:]
+        x[x == 255] = len(types)
```

Comment:
Is 255 returned by `GEOSGeomTypeId` to indicate missing? (not included in the types)

If so, I think you need to put this to -1 (this is used as missing value indicator in codes):

```
In [36]: pd.Categorical.from_codes([0,1,2], ['a', 'b'])
...
ValueError: codes need to be between -1 and len(categories)-1

In [37]: pd.Categorical.from_codes([0,1,-1], ['a', 'b'])
Out[37]: 
[a, b, NaN]
Categories (2, object): [a, b]
```